plugins/woocommerce/includes/shortcodes/class-wc-shortcode-checkout.php

```diff
@@ -382,75 +382,12 @@ private static function checkout() {
-		 * verification in *all* cases).
-		 *
-		 * @since 8.0.0
-		 *
-		 * @param int      $grace_period Time in seconds after an order is placed before email verification may be required.
-		 * @param WC_Order $order        The order for which this grace period is being assessed.
-		 * @param string   $context      Indicates the context in which we might verify the email address. Typically 'order-pay' or 'order-received'.
-		 */
-		$verification_grace_period = (int) apply_filters( 'woocommerce_order_email_verification_grace_period', 10 * MINUTE_IN_SECONDS, $order, $context );
-		$date_created              = $order->get_date_created();
-
-		// We do not need to verify the email address if we are within the grace period immediately following order creation.
-		if (
-			is_a( $date_created, WC_DateTime::class )
-			&& time() - $date_created->getTimestamp() <= $verification_grace_period
-		) {
-			return false;
-		}
-
-		$session       = wc()->session;
-		$session_email = '';
-
-		if ( is_a( $session, WC_Session::class ) ) {
-			$customer      = $session->get( 'customer' );
-			$session_email = is_array( $customer ) && isset( $customer['email'] ) ? $customer['email'] : '';
-		}
-
-		// Email verification is required if the user cannot be identified, or if they supplied an email address but the nonce check failed.
-		$can_view_orders      = current_user_can( 'read_private_shop_orders' );
-		$session_email_match  = $session_email === $order->get_billing_email();
-		$supplied_email_match = sanitize_email( wp_unslash( filter_input( INPUT_POST, 'email' ) ) ) === $order->get_billing_email()
-			&& wp_verify_nonce( filter_input( INPUT_POST, 'check_submission' ), 'wc_verify_email' );
-
-		$email_verification_required = ! $session_email_match && ! $supplied_email_match && ! $can_view_orders;
-
-		/**
-		 * Provides an opportunity to override the (potential) requirement for shoppers to verify their email address
-		 * before we show information such as the order summary, or order payment page.
-		 *
-		 * Note that this hook is not always triggered, therefore it is (for example) unsuitable as a way of forcing
-		 * email verification across all order confirmation/order payment scenarios. Instead, the filter primarily
-		 * exists as a way to *remove* the email verification step.
-		 *
-		 * @since 7.9.0
-		 *
-		 * @param bool     $email_verification_required If email verification is required.
-		 * @param WC_Order $order                       The relevant order.
-		 * @param string   $context                     The context under which we are performing this check.
-		 */
-		return (bool) apply_filters( 'woocommerce_order_email_verification_required', $email_verification_required, $order, $context );
+		// If we cannot match the order with the current user, ask that they verify their email address.
+		$supplied_email = sanitize_email( wp_unslash( filter_input( INPUT_POST, 'email' ) ) ) === $order->get_billing_email()
```

Comment:
This one liner is quite hard to understand at first. Let's take the opportunity and refactor it into a more readable code. Maybe change to something like this:

```
// If we cannot match the order with the current user, ask that they verify their email address.
$supplied_email_matches_order_email = sanitize_email( wp_unslash( filter_input( INPUT_POST, 'email' ) ) ) === $order->get_billing_email();
$nonce_is_valid                     = wp_verify_nonce( filter_input( INPUT_POST, 'check_submission' ), 'wc_verify_email' );
$supplied_email                     = null;

if ( $supplied_email_matches_order_email &&  $nonce_is_valid ) {
	$supplied_email = sanitize_email( wp_unslash( filter_input( INPUT_POST, 'email' ) ) );
}
```

Also it ends up checking if the email matches with the order again in the new function. Do we really need to also check it here?

Reply:
>This one liner is quite hard to understand at first. Let's take the opportunity and refactor it into a more readable code.

Good idea, thank you! 

>Also it ends up checking if the email matches with the order again in the new function. Do we really need to also check it here?

Thank you for catching that!  Both issues are fixed here [cf90234](https://github.com/woocommerce/woocommerce/pull/43834/commits/cf9023489423497b3cb763b055a186df2ffc9c9a).